stock_location_archive_constraint/models/product_product.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    @api.constrains("active")
+    def _check_active(self):
```

Comment:
As there's the possibility of other module introducing the same constraint method, please call it:
```suggestion
    def _check_active_stock_location_archive_constraint(self):
```

Reply:
Well, if the module name changes, then this should change as well.